fog/sql_recovery_db/src/lib.rs

```diff
@@ -1147,14 +1147,12 @@ impl SqlRecoveryDb {
             .load::<(Option<i64>, String, Vec<u8>, i64)>(conn)?
             .into_iter()
             .map(|(ingest_invocation_id, report_id, report, pubkey_expiry)| {
-                let attestation_evidence = AttestationEvidenceOneOf::decode(&*report)?.evidence;
+                let attestation_evidence = EvidenceKind::from_bytes(report)?;
```

Comment:
Since this decodes here and encodes on line 1221 our tests ensure the round trip logic works, but they don't actually check the historically encoded EPID report could be decoded.

I'm not sure it's worth it, updating the tests to support it as EPID should go away after the 6.0 release. We only saw this issue as we tried to re-use a database that had a pre-existing fog report in it.  If we had removed the table entries like the one fog migration it would have been a non issue.

The way this was tracked down was that I ran the `test_reports_db()` test case and replaced the `report` bytes here with the `hex::decode()` output of the attached txt file.

For reference the start of this text file is
```
0a83020a80020
```
while the previous behavior would result in hex bytes of
```
12bd1e0a83020a8
```
Notice the `12bd1e` which represents the proto tag `2` and the size of the subsequent bytes.
[old_report.txt](https://github.com/mobilecoinfoundation/mobilecoin/files/13421914/old_report.txt)